spec/concerns/liquid_interpolatable_spec.rb

```diff
@@ -165,5 +165,18 @@ def @filter.to_xpath_roundtrip(string)
         expect(@agent.interpolated['long_url']).to eq('http://2many.x/6')
       end
     end
+    
+    describe 'regex_replace' do
+      it 'should replace all occurences of a string using regex' do
+        expect(@filter.regex_replace('foobar foobar', /\S+bar/, 'foobaz')).to eq('foobaz foobaz')
+      end
+    end
+
```

Comment:
Might be nice to do something like the above describe group--

```ruby
describe 'regex replace' do
  let(:agent) { Agents::InterpolatableAgent.new(name: "test") }

  it 'should replace all occurrences of a string using regex' do
    agent.interpolation_context['something'] = 'foobar foobar'
    agent.options['cleaned'] = '{{ regex_replace_first(something, /\S+bar/, 'foobar') }}'
    expect(agent.interpolated['cleaned']).to eq('foobaz foobar')
  end
end
```

or something?  I don't think I used your filter correctly there, though.